torchao/dtypes/affine_quantized_tensor.py

```diff
@@ -613,6 +620,163 @@ def from_plain(
         int_data_compressed = torch._cslt_compress(int_data)
         return cls(int_data_compressed, scale, zero_point, layout_type)
 
+@register_layout_cls(BlockSparseLayoutType)
```

Comment:
@jcaip do you want to move these to sparsity folder? now we should be able to do everything outside of affine_quantized_tensor after https://github.com/pytorch/ao/pull/783

(meaning register layout_type/layout tensor and dispatch/impl for quantized linear